lib/iris/tests/idiff.py

```diff
@@ -236,7 +237,17 @@ def step_over_diffs(result_dir, action, display=True):
                         # So copy the local file to the exected file to
                         # maintain this helpfulness.
                         shutil.copy(local_fname, expected_fname)
-                mcompare.compare_images(expected_fname, result_fname, tol=0)
+                try:
+                    mcompare.compare_images(expected_fname, result_fname,
+                                            tol=0)
+                except Exception as e:
+                    if isinstance(e,  ValueError) or \
+                            isinstance(e, ImageComparisonFailure):
+                        print('Could not compare {}: {}'.format(result_fname,
+                                                                e.message))
+                        continue
+                    else:
+                        raise
```

Comment:
No, `raise` on its own will keep the original stack trace. `raise e` would start a new stack trace from here. Minor detail, but it gives an extra bit of debugging information.